### PR TITLE
Performance improvement.

### PR DESCRIPTION
**Pull request by jayy-lmao**
Performance increases inspired by https://jsperf.com/flatten-an-array-loop-vs-reduce/47 and arr-flatten (https://github.com/jonschlinkert/arr-flatten/blob/master/index.js).

See here for performance comparison: https://jsperf.com/unknown-depth-flatten

### CODE REVIEW

**Comment by mk-pmb** [comment]
Thanks! I've added jslint to express at least the machine-checkable parts of what coding style I'm planning to use, would be nice if you could do the rebase for me. Sorry for the trickle of additional change requests, doing a PR in the middle of a maintainer transition is probably rather annoying.

Also consider adding yourself to the contributors in `package.json` if you like, above me. (I'll use "maintainer last, others alphabetically" to reduce diff noise.)

Edit: I had a mistake in the jslint directives, so you might need to rebase yet again. Sorry!

**Comment by jayy-lmao** [comment]
I have changed it for jslint and added me as a contributor :+1:

**Comment by mk-pmb** [comment]
The 2nd commit subject looks like a mistake. Please rewrite the history so that each commit is descriptive. (In this case you'll probably need just one commit for that.)

**Comment by mk-pmb** [comment]
Could you try changing the merge target to the new `experimental` branch? If that doesn't work, I'll just swap the branch heads immediately after merge.

**Comment by jayy-lmao** [comment]
If it's alright I'm just going to open up a new PR into experimental, think that might be easier for both of us.

File: index.js [hunk 1]
```diff
@@ -1,20 +1,19 @@
...
-        return acc.concat(coreFlatten(item, d + 1));
-      }
-      return acc.concat(item);
-    }, []);
+"use strict";
+function flatRecurs(arr, d, res) {
+  var i = 0, cur, len = arr.length;
+  for (i; i < len; i += 1) {
+    cur = arr[i];
+    if (Array.isArray(cur) && d) {
```
Comment (mk-pmb) [inline index.js:8]: What's the `&& d` for? Is there any false-y value that would be accepted by Array.isArray?

Comment (mk-pmb) [reply index.js:8]: Oh, sorry. I didn't read properly. Nevermind.

Comment (mk-pmb) [reply index.js:8]: Then again, could it give better performance to switch them? Checking a boolean value should be cheaper than calling a function, but then again it depends on which of both checks will fail more often, and that depends on input.

Comment (jayy-lmao) [reply index.js:8]: I've swapped them in any case. Going to close this PR and open a new one into experimental.

**Pull request closed without merging**